ws/ws.go

```diff
@@ -99,7 +100,6 @@ func Write(key *datastore.Key, buf []byte) error {
 }
 
 func reader(ws *websocket.Conn) {
-	defer ws.Close()
```

Comment:
Have you tried logging whether the connection was closed, to be sure that this is the cause?

````go
defer func() {
	// Log "Closing WebSocket".
	ws.Close()
}
````